### ISSUE TITLE
ModuleNotFoundError Received when running sls offline for a Python project.

### ISSUE THREAD

**Issue by ezzy1337** [SamWSoftware/serverless-auto-swagger#10]
I didn't see it explicitly stated anywhere but does this plugin work for a serverless application that is not typescript?

Given the error I'm seeing it looks like Python is trying to import the `html.js` file from the pregenerated `swagger` folder which is an invalid import. 

```
Traceback (most recent call last):
  File "/Users/derek/platform-api/services/checkpoint-service/node_modules/serverless-offline/dist/lambda/handler-runner/python-runner/invoke.py", line 76, in <module>

    module = import_module(args.handler_path.replace(os.sep, '.'))
  File "/usr/local/Cellar/python@3.9/3.9.0_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 984, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'swagger.html'
```

Here is the configuration I've provided for the plugin.
```
  autoswagger:
    generateSwaggerOnDeploy?: true
    typefiles?: [ './schemas/checkpoint.d.ts' ]
```

I added the `typefiles?` portion and the checkpoint.d.ts file just to see if it resolved the import but to no avail.

Other Pertinent Details:
```
  Your Environment Information ---------------------------
     Operating System:          darwin
     Node Version:              15.0.1
     Framework Version:         2.71.0 (local)
     Plugin Version:            5.5.3
     SDK Version:               4.3.0
     Components Version:        3.18.1
```

**Comment by SamWSoftware** [comment]
I've never tested it with a python project. The endpoint that it makes by default is a Javascript one. Not sure how that will work with a python serverless project.

If there was a way to check if the project was js or python then the plugin could swap out the html.js for an equivalent python lambda

**Comment by ezzy1337** [comment]
I think we could use the `runtime` field under the `provider` section in the serverless.yml file to determine if it's a javascript or python project. 

Over the weekend I played around with a basic python serverless app. The basic python app would get the same error `ModuleNotFoundError: No module named 'swagger.html'`. but if I changed the `runtime` from `python3.8` to `nodejs12.x` the swagger endpoints would work.

If you can point me in the right direction I'd be happy to work on this.

For reference here is the serverless.yml file I was working with.

```
org: ddrummond
app: serverless-docs-python
service: serverless-docs-python

frameworkVersion: '2 || 3'

provider:
  name: aws
  runtime: python3.8  # <-- I think we could use the 
  lambdaHashingVersion: 20201221

functions:
  health:
    handler: handler.health
    events:
      - http:
          method: get
          path: /health
          cors: true

plugins:
  - serverless-auto-swagger
  - serverless-offline
```

**Comment by ezzy1337** [comment]
Another thought is adding a projectRuntime option to the custom config for the plugin.

**Comment by SamWSoftware** [comment]
PR approved and merged in. Just publishing a new version (1.4.0) now so let me know if it works with that

**Comment by ezzy1337** [comment]
@SamWSoftware Thanks. I think I need to follow-up with another PR to fix a small issue with the swagger json written from the api-types.d.ts files. They write `true` and `false` for some fields and Python needs `True` and `False`. You might want to hold up on a release until I get that fixed as well.

**Comment by fcrdossantos** [comment]
Hi @ezzy1337 and @SamWSoftware i'm facing the same problem

Even after @ezzy1337 PR's the error continues. 

Debugging it i noticed 2 erros:

1) names=>  When you try to run "html.handler" and "json.handler" (html.py and json.py from #12 ) python tries to import the build-in modules

html https://docs.python.org/3/library/html.html
and 
json https://docs.python.org/3/library/json.html

If i change the names to something like "htmlpy" and "jsonpy" it's start to find the correct files to import.


2) docs importation => the file "json.py" tries to do this:

```from .swagger import docs```

but cames to an error (": "attempted relative import with no known parent package",) 

to get it works you need to do this:

```from swagger import docs```

(just removed the dot "." before swagger) 


Can you guys please provide a fix for it? :)

**Comment by ezzy1337** [comment]
I must have missed the release for my changes. It sounds like you have a good grasp on the changes needed might be a great time to become an open source contributor. 

I think the problem could be resolved by adding an `__init__.py` file to the directory with the html.py and json.py files. Overwriting a built in namespace is not ideal though. Even if relative imports are used  I'd have to look into that more.

**Comment by fcrdossantos** [comment]
Just to fix it for me i changed some files (but my solution was made in a very wrong way, so i didnt sent as a PR hahaha)

on `resources/functions.js`:

Line 13: `handler: handlerPath + 'htmlpy.handler',`
Line 27: `handler: handlerPath + 'jsonpy.handler',`

Basically, i changed the names, so Python dont try to import the built-in "html" and "json" modules.

on `resources/` folder:

Renamed from `"html.py"` to `"htmlpy.py"`
Renamed from `"json.py"` to `"jsonpy.py"`

On `"json.py"` (now "jsonpy.py"):

Line 2: `from swagger import docs`


@SamWSoftware i was thinking about renaming all the files under resources folder using a prefix, to something like:

"swagger_html.py" | "swagger_html.js" | "swagger_html.d.ts" {...}
"swagger_json.py" | "swagger_json.js" | "swagger_json.d.ts" {...} 

Doing that we solve the naming conflict on Python and then we just need to change the Line 2 on "swagger_json.py"

But as it's your naming convention, i preefer to talk with you first and see what you think.

If the name change is ok for you, this can help a lot of Python Developers :)

Btw, thanks for you contributtion, serverless-auto-swagger is awesome!

**Comment by fcrdossantos** [comment]
> I must have missed the release for my changes. It sounds like you have a good grasp on the changes needed might be a great time to become an open source contributor.
> 
> I think the problem could be resolved by adding an `__init__.py` file to the directory with the html.py and json.py files. Overwriting a built in namespace is not ideal though. Even if relative imports are used I'd have to look into that more.

Even if you create the __init__.py you will received errors on json import for 2 reasons:
1) Lambda tries to import built-in json on handler invocation 
2) Inside /resources/json.py you try to import json, then python will try to "re-import" the file

I tried both solutions early and the one that really works was changing the names =/

**Comment by fcrdossantos** [comment]
@SamWSoftware i oppened a new PR #17  with those Fixes (with a better solution) and new features, if you can take a look :)

I don't have time to maintain a fork, but a think that i can help you with this PR, feel free to change my code as you desire :)

**Comment by bfaulk96** [comment]
@ezzy1337 This should be fixed in `1.6.0` and beyond, could you please let me know if not?
